asf_core_data/pipeline/mcs/process/process_historical_mcs_installers.py

```diff
@@ -0,0 +1,926 @@
+    missing_installers[["full_address", "date_of_creation"]] = missing_installers.apply(
+        lambda x: match_companies_house(
+            x["installation_company_name"], companies_house_api_key
+        ),
+        axis=1,
+    )
+
+    # date_of_creation type change: str -> datetime
+    missing_installers["date_of_creation"] = missing_installers[
+        "date_of_creation"
+    ].apply(lambda x: datetime.strptime(x, "%Y-%m-%d") if not pd.isnull(x) else x)
+
+    # We only keep address info if the company creation date is before the first comissioning date (serves as a check)
+    start_date_cols = [col for col in missing_installers if "start_date" in col]
+    missing_installers["first_commissioning_date"] = missing_installers[
+        start_date_cols
+    ].min(axis=1)
+
+    missing_installers["full_address"] = missing_installers.apply(
+        lambda x: recompute_full_adress(
+            x["date_of_creation"], x["first_commissioning_date"], x["full_address"]
+        ),
+        axis=1,
+    )
+
+    # extract address_1 and postcode from full_address variable
+    missing_installers["address_1"] = missing_installers["full_address"].apply(
+        lambda x: x.split(",")[0] if x is not None else None
+    )
+    missing_installers["postcode"] = missing_installers["full_address"].apply(
+        lambda x: x.split(",")[-1].strip() if x is not None else None
+    )
+
+    # rename columns to match installers table
+    missing_installers.rename(
+        {
+            "installation_company_name": "company_name",
+            "installation_company_mcs_number": "mcs_certificate_number",
+        },
+        axis=1,
+        inplace=True,
+    )
+
+    # dropping unecessary variables
+    missing_installers.drop(
+        ["date_of_creation"],
+        axis=1,
+        inplace=True,
+    )
+
+    return missing_installers
+
+
+def update_full_address(
+    address_1: str,
+    address_2: str,
+    town: str,
+    county: str,
+    postcode: str,
+    original_record: str,
+    full_address: str,
+) -> str:
+    """
+    Computes full_address by putting together the different lines in address
+    (address_1, address_2, town, county and postcode) for original records.
+
+    Args:
+        address_1: first line of address
+        address_2: second line of address
+        town: town
+        county: county
+        postcode: postcode
+        original_record: True if record originally in installers table
+    Returns:
+        full adress
+    """
+    if original_record:
+        temp = address_1
+        if not pd.isnull(address_2):
+            temp = temp + address_2 + ", "
+        if not pd.isnull(town):
+            temp = temp + town + ", "
+        if not pd.isnull(county):
+            temp = temp + county + ", "
+        if not pd.isnull(postcode):
+            temp = temp + postcode + ", "
+        return temp[:-2]
```

Comment:
That looks fine, but you could also try something like to prevent repetition and confusion about `temp[:-2]`

```suggestion
    if original_record:
        temp = []
        for address_part in [address_1, address_2, town, county, postcode]:
              if not pd.isnull(address_part):
                    temp.append(address_part)
        full_address = ", ".join(temp)
        return full_address
 
```

Reply:
very neat! will add :)